op-proposer/proposer/driver.go

```diff
@@ -480,12 +481,19 @@ func (l *L2OutputSubmitter) loopL2OO(ctx context.Context) {
 			// A note on retrying: the outer ticker already runs on a short
 			// poll interval, which has a default value of 6 seconds. So no
 			// retry logic is needed around output fetching here.
-			output, shouldPropose, err := l.FetchL2OOOutput(ctx)
+			var output *eth.OutputResponse
+			var shouldPropose bool
+			var err error
+			if l.dgfContract == nil {
+				output, shouldPropose, err = l.FetchL2OOOutput(ctx)
+			} else {
+				output, shouldPropose, err = l.FetchDGFOutput(ctx)
```

Comment:
I noticed that we never return a non-nil `output` when `shouldPropose=false`. Since `output` is a pointer, we could consider using the `nil` value to stand for "don't propose".

This might be an optimization too far / question of taste. Don't feel hugely strongly.

Reply:
That's certainly possible. Idiomatic Go likes to be more explicit and not use `nil` values to encode other meaning, so I suggest to keep the bool for now.